pkg/controller/operators/olm/operatorconditions.go

```diff
@@ -0,0 +1,35 @@
+package olm
+
+import (
+	"context"
+	"fmt"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorv1 "github.com/operator-framework/api/pkg/operators/v1"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+)
+
+func (a *Operator) isOperatorUpgradeable(csv *v1alpha1.ClusterServiceVersion) (bool, error) {
+	if csv == nil {
+		return false, fmt.Errorf("CSV is invalid")
+	}
+
+	cond, err := a.lister.OperatorsV1().OperatorConditions(csv.GetNamespace()).Get(context.TODO(), csv.GetName(), metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return true, nil
+		}
+		return false, err
+	}
+
+	if len(cond.Status.Conditions) < 1 {
```

Comment:
@dinhxuanvu @awgreene Just want you to sanity check the interaction of these two parts.

It looks like:

1. An OperatorCondition is generated for every operator installed
2. There's a requirement that the condition be set to upgradeable=true in order to install successfully

Doesn't that mean that for operators that don't support / know how to update an OperatorCondition, their CSV's will never go to succeeded?

Reply:
This if statement returns `true`.

Reply:
And what ensures that an operator has a chance to set `Upgradeable=false` before this check?

Reply:
We can use the timeStamp for this purpose.

Reply:
Synced offline - 

There will be an expectation that any operator that opts into this feature will update the OperatorCondition before their readiness probe returns true.